### PR TITLE
Backport 4438 to 2.2.0

### PR DESCRIPTION
**Pull request by anchitj**
Removed 0143 test references.

Also backported #4775, #4872, #4794 to make CI pass.

Backported #4726, #4706, #4875 to upgrade linux dependencies.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,7 +25,8 @@ librdkafka v2.2.0 is a feature release:
    Add DNS alias support for secured connection (#4292).
  * [KIP-339](https://cwiki.apache.org/confluence/display/KAFKA/KIP-339%3A+Create+a+new+IncrementalAlterConfigs+API):
    IncrementalAlterConfigs API (started by @PrasanthV454, #4110).
- * [KIP-554](https://cwiki.apache.org/confluence/display/KAFKA/KIP-554%3A+Add+Broker-side+SCRAM+Config+API): Add Broker-side SCRAM Config API (#4241).
+ * [KIP-554](https://cwiki.apache.org/confluence/display/KAFKA/KIP-554%3A+Add+Broker-side+SCRAM+Config+API): Add Broker-side SCRAM Config API (#4241). 
+ * Fix for idempotent producer fatal errors, triggered after a possibly persisted message state (#4438).
```
Comment (emasab) [inline CHANGELOG.md:29]: The fix should go in a 2.2.1 maintenance release with only this item

Comment (emasab) [reply CHANGELOG.md:29]: Is customer using Windows only? Otherwise we should upgrade Linux dependencies too.

Comment (emasab) [reply CHANGELOG.md:29]: `RD_KAFKA_VERSION` must be changed to `020201` and version in `vcpkg.json` must be `2.2.1` too.

**Review by emasab — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (anchitj) [reply CHANGELOG.md]: Moved the changelog to v.2.2.1 and updated `RD_KAFKA_VERSION`

Comment (emasab) [reply CHANGELOG.md]: In `rdkafkacpp.h` too, please

**Comment by anchitj** [comment]
Updated in `rdkafkacpp.h`

**Review by emasab — APPROVED** [review]

**Pull request merged**